### PR TITLE
Fixed issues in fhirr4 and ci workflows

### PR DESCRIPTION
**Pull request by GevinHasmitha**
1) Added separate workflow for the fhirr4 as it uses maven build.
2) Changed the ci workflow so that it works with the maven build (it uses maven build for fhirr4 only and uses bal build for other packages). 
3) Added code so that the versions in the pom files of fhirr4 are updated instead of the version in ballerina.toml.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -77,13 +80,45 @@ jobs:
...
+      - name: Setup Maven
+        if: ${{ matrix.path == 'fhirr4/ballerina/src/main/resources/fhirservice' }}
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+
+      - uses: ballerina-platform/setup-ballerina@v1.1.0
+        if: ${{ matrix.path == 'fhirr4/ballerina/src/main/resources/fhirservice' }}
+        name: Setup latest Ballerina version
```
Comment (ThishaniLucas) [inline .github/workflows/ci.yml:100]: Lets bring the name of the task up.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -77,13 +80,45 @@ jobs:
...
+                  <password>${{ secrets.PAT_TOKEN }}</password>
+                </server>
+              </servers>
+            </settings>' > ~/.m2/settings.xml
+
+      - name: Run maven build
+        if: ${{ matrix.path == 'fhirr4/ballerina/src/main/resources/fhirservice' }}
+        run: |
+          cd fhirr4
+          mvn clean install
```
Comment (ThishaniLucas) [inline .github/workflows/ci.yml:120]: No need to cd. Diretory name can be passed to mvn command with -f flag.

File: .github/workflows/fhirr4-build-executor.yml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Setup Java and Maven
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+
+      - uses: ballerina-platform/setup-ballerina@v1.1.0
+        name: Setup latest Ballerina version
```
Comment (ThishaniLucas) [inline .github/workflows/fhirr4-build-executor.yml:28]: Bring the name up

File: .github/workflows/fhirr4-build-executor.yml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+              JAVA_HOME: /usr/lib/jvm/default-jvm
+              BALLERINA_STAGE_CENTRAL: true
+              BALLERINA_CENTRAL_ACCESS_TOKEN: ${{ secrets.BALLERINA_CENTRAL_STAGE_ACCESS_TOKEN }}
+
+      - name: Run ballerina build for dev
+        if: inputs.bal_central_environment == 'DEV'
+        uses: ballerina-platform/ballerina-action@2201.7.0
+        with:
+          args:
+            pack
```
Comment (ThishaniLucas) [inline .github/workflows/fhirr4-build-executor.yml:68]: fhirr4 doesn't require bal pack. It's done via mvn.

**Review by ThishaniLucas — CHANGES_REQUESTED** [review]
Please go through the readme of the repo and see how fhirr4 is built and published.

File: .github/workflows/fhirr4-build-executor.yml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+          packageName=$(basename ${{ inputs.working_dir }})
+
+          # Commit changes
+          git config --global user.name ${{ secrets.BALLERINA_BOT_USERNAME }}
+          git config --global user.email ${{ secrets.BALLERINA_BOT_EMAIL }}
+          git add ${{ inputs.working_dir }}/pom.xml
+          git add ${{ inputs.working_dir }}/ballerina/pom.xml
+          git add ${{ inputs.working_dir }}/compiler-plugin/pom.xml
+          git add ${{ inputs.working_dir }}/native/pom.xml
+          git add ${{ inputs.working_dir }}/Dependencies.toml
```
Comment (ThishaniLucas) [inline .github/workflows/fhirr4-build-executor.yml:137]: No need to commit dependencies toml

File: .github/workflows/fhirr4-build-executor.yml [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+          IFS='.' read -ra VERSION_PARTS <<< "$CURRENT_VERSION"
+          PATCH_VERSION=$((VERSION_PARTS[2] + 1))
+          NEW_VERSION="${VERSION_PARTS[0]}.${VERSION_PARTS[1]}.$PATCH_VERSION"
+          echo "NEW_VERSION=${NEW_VERSION}" >> $GITHUB_OUTPUT
+          echo "CURRENT_VERSION=${CURRENT_VERSION}" >> $GITHUB_OUTPUT
+          #Bump the versions in each pom file in fhirr4
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/compiler-plugin/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/native/pom.xml
+          sed -i "0,/<version>${CURRENT_VERSION}<\/version>/s//<version>${NEW_VERSION}<\/version>/" ${{ inputs.working_dir }}/ballerina/pom.xml
```
Comment (ThishaniLucas) [inline .github/workflows/fhirr4-build-executor.yml:122]: This is fine for now. But as a future improvement let's use an XML processing tool to do this.

Comment (GevinHasmitha) [reply .github/workflows/fhirr4-build-executor.yml:122]: Got it👍

**Review by ThishaniLucas — APPROVED** [review]

**Pull request merged**